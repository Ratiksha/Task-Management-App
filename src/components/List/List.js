import React, { Component } from 'react';
import './List.scss';
import Card from '../Card/Card';
import Modal from '../Modal/Modal'
import AddTaskForm from '../AddTaskForm/AddTaskForm'
import DeleteForm from '../DeleteForm/DeleteForm'
import AddListForm from '../AddListForm/AddListForm'
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrashAlt, faPencilAlt } from "@fortawesome/free-solid-svg-icons";


export default class List extends Component{
    constructor(props){
        super(props);
        this.state = {
            showModal: false,
            showEditForm: false,
            showDeleteForm: false
        }
        this.handleKeyPress = this.handleKeyPress.bind(this);
    }

    componentDidMount(){
        document.addEventListener('keydown', this.handleKeyPress);
    }

    handleKeyPress(event) {
        if (event.keyCode === 13) {
            document.activeElement.click();
        }
      }

    showModal = (formType) => {
        this.toggleModal();
        if(formType === 'edit'){
            this.setState({ 
                showEditForm: true,
                showDeleteForm: false
            });
        }
        else if(formType === 'delete') {
            this.setState({ 
                showDeleteForm: true,
                showEditForm: false
            });
        }
    }
    
    hideModal = () => {
        this.toggleModal();
    }

    toggleModal = () => {
        this.setState({ 
            showModal: !this.state.showModal,
            showDeleteForm: false,
            showEditForm: false
         });
    }

    render() {
        const { lists, list, addTask, editTask, DeleteTask, editList, DeleteList, onTaskDragStart, onTaskDragOver, onTaskDrop, onListDragStart  } = this.props;
        const cards = list && list.cards.map((card, index) => (
            <li key={index}>
                <Card lists={lists} card={card} editTask={editTask} DeleteTask={DeleteTask} listId={list.listId} onTaskDragStart={onTaskDragStart}/>
            </li>
        ));

        return (
            <div draggable="true" id={[ list && list.listId]} onDragStart={onListDragStart}>
                <h3 className="header">{list && list.name}
                <div className="button-group">
                    <button id="list-edit" onClick={() => this.showModal('edit')}><FontAwesomeIcon icon={faPencilAlt}/></button>
                    <button id="list-delete" onClick={() => this.showModal('delete')}><FontAwesomeIcon icon={faTrashAlt}/></button>
                </div>
                </h3>
                <ul className="list" onDragOver={onTaskDragOver} onDrop={onTaskDrop}>
                    {cards}
                    <li>
                        <AddTaskForm lists={lists} listId={list && list.listId} addTask={addTask} hideModal={this.hideModal}/>
                    </li>
                </ul>
                <Modal show={this.state.showModal} handleClose={this.hideModal}>
                    {this.state.showEditForm && <AddListForm lists={lists} listId={list && list.listId} editList={editList} name={list && list.name} hideModal={this.hideModal}/>}
                   {this.state.showDeleteForm && <DeleteForm listId={list && list.listId} DeleteList={DeleteList} name={list && list.name} hideModal={this.hideModal}/>}
                </Modal>
            </div>
        )
    }
}