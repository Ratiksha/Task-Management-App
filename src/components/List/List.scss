@import '../variables.scss';
.header{
    padding: 8px 10px;
    margin: 0;
    position: relative;
    background: var(--color);
    color: $white;
    border-top-right-radius: 5px;
    border-top-left-radius: 5px;
    .button-group{
        position: absolute;
        top: 7px;
        right: 10px;
        button{
            background-color: transparent;
            padding: 0;
        }
        svg{
            color: $white;
            margin: 5px;
            font-size: 10px;
            cursor: pointer;
        }
    }
}

.list{
    list-style-type: none;
    padding: 20px 8px 20px;
    border-radius: 2px;
    min-height: 218px;
}