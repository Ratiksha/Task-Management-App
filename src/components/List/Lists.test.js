import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

let wrapped = shallow(<List />);
let editType = 'edit';
let deleteType = 'delete';

describe('List', () => {
  it('should render the List Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  it('When edit button is clicked', () => {
    const editButton = wrapped.find('#list-edit');
    editButton.simulate('click');
    expect(editType).toBe('edit');
  });

  it('When delete button is clicked', () => {
    const deleteButton = wrapped.find('#list-delete');
    deleteButton.simulate('click');
    expect(deleteType).toBe('delete');
  });
});