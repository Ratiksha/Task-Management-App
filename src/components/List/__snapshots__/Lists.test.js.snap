// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render the List Component correctly 1`] = `ShallowWrapper {}`;
